jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/ee/rs/cookieparam/locator/JAXRSLocatorClientIT.java

```diff
@@ -59,21 +63,33 @@ public class JAXRSLocatorClientIT
   private static final long serialVersionUID = 1L;
 
   public JAXRSLocatorClientIT() {
-    setup();
     setContextRoot("/jaxrs_ee_rs_cookieparam_locator_web/resource/locator");
   }
 
+  @ArquillianResource
+  @OperateOnDeployment("jaxrs_ee_rs_cookieparam_locator_web")
+  private URL url;
+
   @BeforeEach
-  void logStartTest(TestInfo testInfo) {
-    TestUtil.logMsg("STARTING TEST : "+testInfo.getDisplayName());
-  }
+  public void setup() {
+
+    TestUtil.logTrace("setup method JAXRSLocatorClientIT");
+
+    assertFalse((url==null), "[JAXRSLocatorClientIT] 'url' was not injected.");
+    
+    String hostname = url.getHost();
+    String portnum = Integer.toString(url.getPort());
```

Comment:
This pattern is in several places and I wonder if it should be simplified. Instead of converting it to a string, then back to an integer, we could just use the integer and check if the port is > 0. The `isNullOrEmpty(portnum)` will never be false as the `Integer.toString()` will never return an empty string or `null`.

Reply:
Agree, latest commit resolves it.